apps/src/templates/instructions/TeacherFeedbackKeepWorking.jsx

```diff
@@ -16,56 +16,72 @@ const ReviewStates = makeEnum('completed', 'keepWorking');
 class TeacherFeedbackKeepWorking extends Component {
   static propTypes = {
     latestFeedback: PropTypes.object,
-    reviewState: PropTypes.oneOf(Object.values(ReviewStates)),
     setReviewState: PropTypes.func,
     setReviewStateChanged: PropTypes.func
   };
 
   checkbox = null;
+
   isAwaitingTeacherReview =
+    this.props.latestFeedback &&
     this.props.latestFeedback.review_state === ReviewStates.keepWorking &&
     this.props.latestFeedback.student_updated_since_feedback;
 
+  initialReviewState = this.isAwaitingTeacherReview
+    ? ReviewStates.awaitingReview
+    : this.props.latestFeedback?.review_state || null;
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      reviewState: this.initialReviewState
+    };
+  }
+
   componentDidMount() {
-    if (this.isAwaitingTeacherReview) {
+    this.setCheckboxState();
+  }
+
+  setCheckboxState = () => {
+    if (this.state.reviewState === ReviewStates.awaitingReview) {
       this.checkbox.indeterminate = true;
+    } else {
+      this.checkbox.checked =
+        this.state.reviewState === ReviewStates.keepWorking;
     }
-  }
+  };
 
   onCheckboxChange = () => {
-    const newReviewState = this.getNewReviewState();
-    this.props.setReviewState(newReviewState);
+    const newReviewState = this.getNextReviewState();
+    this.setState({reviewState: newReviewState}, this.setCheckboxState);
 
-    const isDifferentFromInitial = this.isDifferentFromInitial(newReviewState);
-    this.props.setReviewStateChanged(isDifferentFromInitial);
+    this.props.setReviewState(newReviewState);
+    this.props.setReviewStateChanged(
+      newReviewState !== this.initialReviewState
+    );
   };
 
-  getNewReviewState() {
-    let newReviewState = null;
-
-    if (this.checkbox.checked) {
-      newReviewState = ReviewStates.keepWorking;
-    } else if (this.isAwaitingTeacherReview) {
-      newReviewState = ReviewStates.completed;
+  getNextReviewState() {
+    if (this.state.reviewState === ReviewStates.awaitingReview) {
+      return ReviewStates.completed;
+    } else if (this.state.reviewState === ReviewStates.keepWorking) {
+      if (!this.initialReviewState) {
```

Comment:
(very minor) Maybe make L69-73 a ternary?  (Sorry, I just realized that you might have left this expanded because of an unclear comment that I made.)